sdk/core/core/src/az_aad.c

```diff
@@ -96,7 +102,7 @@ AZ_NODISCARD az_result _az_aad_request_token(az_http_request * ref_request, _az_
       .p_policies = {
         { .process = az_http_pipeline_policy_retry, .p_options = NULL },
         { .process = az_http_pipeline_policy_logging, .p_options = NULL },
-        { .process = az_http_pipeline_policy_transport, .p_options = NULL },
+        { .process = az_http_pipeline_policy_transport, .p_options = http_client },
```

Comment:
Let's do this: define a transport policy options struct similar to what we have for some other policies. Put the http_client as the only field in this options structure. The, this p_options is a pointer to the that options structure since, in the future, the structure might get more fields in it. 
(this is why I had ** here; I was thinking about the future of having a struct and keeping it consistent with ALL the other policies).